src/onnx/parse_qlinearconcat.cpp

```diff
@@ -0,0 +1,134 @@
+#include <migraphx/onnx/broadcast_qdq.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/stringutils.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+/*
+ *********************************************************************************
+ *  Reference: see QLinearConcat in                                              *
+ *  https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md  *
+ *********************************************************************************
+
+com.microsoft.QLinearConcat
+Concatenate a list of tensors into a single tensor.All input tensors must have the same shape,
+except for the dimension size of the axis to concatenate on.
+
+Version
+This version of the operator has been available since version 1 of the 'com.microsoft' operator set.
+
+Attributes
+axis : int (required)
+Which axis to concat on
+
+Inputs (3 - ∞)
+Y_scale : TF
+Y's scale.
+
+Y_zero_point : T8
+Y's zero point.
+
+inputs (variadic, heterogeneous) : TV
+List of tensors/scale/zero_point for concatenation
+
+Outputs
+Y : T8
+Concatenated tensor
+
+Type Constraints
+T8 : tensor(uint8), tensor(int8)
+Constrain input and output types to 8 bit signed and unsigned tensors.
+
+TF : tensor(float)
+Constrain scale types to any float tensor type.
+
+TV : tensor(uint8), tensor(int8), tensor(float)
+Sequence of (Tensor, Scale, ZeroPoint) tuples. The type is sequence of (T8, TF, T8).
+*/
+
+struct parse_qlinearconcat : op_parser<parse_qlinearconcat>
+{
+    std::vector<op_desc> operators() const { return {{"QLinearConcat"}}; }
+
+    // basic type checking for QLinearConcat Operator
+    void check_inputs(const std::vector<instruction_ref>& args) const
+    {
+        auto args_size = args.size();
+        // at least 5 input tensors, can be 5, 8, 11 ...
+        if((args_size < 5) or ((args_size - 2) % 3 != 0))
+            MIGRAPHX_THROW("QLINEARCONCAT: missing inputs");
+
+        auto t0_shape = args[2]->get_shape();
+        auto t0_type  = t0_shape.type();
+        for(auto idx = 2; idx < args.size(); idx += 3)
+        {
+            if((args[idx]->get_shape().type() != t0_type) or
+               (args[idx + 2]->get_shape().type() != t0_type))
+            {
+                MIGRAPHX_THROW("QLINEARCONCAT: mismatching types");
+            }
+        }
+    }
+
+    instruction_ref parse(const op_desc& /* opd */,
+                          const onnx_parser& parser,
+                          const onnx_parser::node_info& info,
+                          const std::vector<instruction_ref>& args) const
+    {
+        check_inputs(args);
+        if(not contains(info.attributes, "axis"))
+            MIGRAPHX_THROW("QLINEARCONCAT: missing axis attribute");
+
+        auto axis = parser.parse_value(info.attributes.at("axis")).template at<float>();
+        std::vector<instruction_ref> tmp;
+        for(auto idx = 2; idx < args.size(); idx += 3)
+        {
+            auto l       = args[idx];
```

Comment:
Suggestion: `l` is a very confusing name, right next to a `1`. And what does it mean, please?

Reply:
changed it to a more dscriptive name.